server/server.go

```diff
@@ -2930,10 +2971,9 @@ func (s *Server) readyForConnections(d time.Duration) error {
 	}
 	chk := make(map[string]info)
 
-	end := time.Now().Add(d)
 	for time.Now().Before(end) {
-		s.mu.RLock()
-		chk["server"] = info{ok: s.listener != nil, err: s.listenerErr}
+		s.mu.Lock()
```

Comment:
You need to change that to an RLock(). This has changed in main so that is causing issue now with your PR.

Reply:
Fixed!